### DISCUSSION TITLE
Image size definitions and where used

### DISCUSSION THREAD

**Discussion by torvista** [zencart/zencart#5846] [General]
As I was confused, and the Docs are needing input
https://docs.zen-cart.com/user/images/image_configuration/
I reviewed the constants and their uses.
Results:
https://docs.google.com/spreadsheets/d/1GNSe8Bk5QGeNhLZ4yMqO6Vl-xtyv9wO3auY_uPZRb80/edit?usp=sharing

**Comment by lat9** [comment]
Nice one, @torvista!  Thanks for the details analysis.

**Comment by torvista** [comment]
One could discuss about separating admin image control from storefront, and removing the unused constants.

**Comment by scottcwilson** [comment]
I would be eternally grateful (well, to the end of the week at least) if you would resolve the FIXME in this doc page. :)

**Comment by torvista** [comment]
Nothing in there to be fixed first....?

**Comment by scottcwilson** [comment]
When I say "resolve" I mean "resolve by fixing what's wrong" (not just removing the FIXME line).  :)

**Reply by torvista** [reply]
I meant fixing what's wrong with the image definitions: a pair are not used or should be...

**Reply by scottcwilson** [reply]
- Unused constants - remove from install script, other files is sufficient for now. 
- Documentation - update as needed to correct.

Please don't pursue "separating admin image control from storefront."

**Comment by torvista** [comment]
As removing constants is a db change, the references in the new install can go now, but the removal from existing sites needs to wait until a next version when there's an upgrade script.

**Reply by drbyte** [reply]
No constants to be removed anymore. They were temporarily missing in v1.5.8, but are restored in v2.0.0.
https://github.com/zencart/zencart/pull/6251

**Comment by torvista** [comment]
Product Listings:

Also Purchased: SMALL_IMAGE_WIDTH, SMALL_IMAGE_HEIGHT
Featured Products: IMAGE_FEATURED_PRODUCTS_LISTING_WIDTH, IMAGE_FEATURED_PRODUCTS_LISTING_HEIGHT
Featured Categories: IMAGE_PRODUCT_LISTING_WIDTH , IMAGE_PRODUCT_LISTING_HEIGHT
New: IMAGE_PRODUCT_NEW_WIDTH, IMAGE_PRODUCT_NEW_HEIGHT
Specials: SMALL_IMAGE_WIDTH, SMALL_IMAGE_HEIGHT
Upcoming: no image/table listing. Should have images I think.

So an inconsistent mixture.

Ideally I would suggest 
1. That each have it's own constants...but that they should use SMALL_IMAGE_WIDTH, SMALL_IMAGE_HEIGHT as a default if nothing set in their specific constants, to reduce the multiple settings in admin-images.

1. That all the specific values be defined or not in **one** selection infobox where all these are listed, instead of all those bewildering separate options. So the whole lot can be set/viewed in one operation.

These are not used at all:
![2025-01-29 12_16_36-Admin Configuration Images — Mozilla Firefox](https://github.com/user-attachments/assets/2e696ca2-83bb-41e0-9a92-da8320b4f86c)
